dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataset/SectionService.java

```diff
@@ -84,4 +85,15 @@ public interface SectionService {
    * @return a Collection of Sections.
    */
   List<Section> getAllSections();
+
+  /**
+   * Retrieves all {@link Section}s referencing the provided {@link Indicator}s.
+   *
+   * @return a Collection of {@link Section}s.
+   */
+  List<Section> getSectionsByIndicators(List<Indicator> indicators);
```

Comment:
Could we reference the indicators we look for by `UID`, so `List<UID>`?
Alternatively, I think we should not always go through a service to use the store. It is ok for the merge service to use other stores. Just if something is a service layer API it should avoid accepting mapped entities as parameters.

Reply:
The underlying store call uses the `Indicator`s to retrieve items from the mapping table `sectionindicators`.
If we were to use `UID`s here then we would have to get the full `Indicator` objects again before making the store call. (basically going from `Indicator` -> `UID` -> `Indicator` which doesn't seem worth it).
Can chat about it if it's not clear.

Reply:
I can update to use the store directly here though.